Development/nmos/api_downgrade.cpp

```diff
@@ -1,11 +1,973 @@
 #include "nmos/api_downgrade.h"
 
 #include <map>
+#include "cpprest/json_validator.h"
+#include "cpprest/basic_utils.h"
 #include "nmos/is04_versions.h"
 #include "nmos/resource.h"
+#include "nmos/type.h"
 
 namespace nmos
 {
+    namespace details
+    {
+        static web::json::value make_schema(const char* schema)
+        {
+            return web::json::value::parse(utility::s2us(schema));
+        }
+
+        static web::uri make_schema_id(const api_version& version, const nmos::type& type, const utility::string_t& property)
+        {
+            return U("/downgrade") + make_api_version(version) + U("-") + type.name + U("-") + property;
+        }
+
+        // property schemas are extracted from the resource schemas
```

Comment:
Since all of these property schemas are already included in the full schemas embedded in nmos-cpp, could this code be refactored to use a json value path to extract the property value schema from those? Not sure...

Reply:
If we agree with this PR, this can be followed up.